### PR TITLE
Make subclasses have their own drop target.

### PR DESCRIPTION
**Pull request by ryan-rushton**
This adds a new d2 bucket category so that subclasses have their own setup. By making `Class` its own bucket category this also fixes the issue with the subclass being populated when filling general items from equipped.

As a side effect subclasses will have their own section too.

### Behaviour

https://user-images.githubusercontent.com/7344652/154796679-b6e0f7c4-6c1f-4852-abea-2b3e63f8253b.mov

### New section in inventory

Screen Shot 2022-02-19 at 8 07 45 pm

### CODE REVIEW

File: src/app/loadout/loadout-edit/LoadoutEditBucketDropTarget.m.scss [hunk 1]
```diff
@@ -29,4 +29,5 @@
 // items entirely.
 .dragOver {
   pointer-events: none;
+  opacity: 0;
```
Comment (ryan-rushton) [inline src/app/loadout/loadout-edit/LoadoutEditBucketDropTarget.m.scss:32]: This just hides the current section when we are showing the equipped/unequipped drop targets. It looked really weird that you could see everything underneath the two button things. I added a part to the video where to show what happens when you drap something like armour

**Comment by ryan-rushton** [comment]
Already ticked off the item in #7841

**Comment by nev-r** [comment]
i'm pro-being able to collapse the section, but this adds a completely new tab to the bottom of mobile

**Comment by ryan-rushton** [comment]
Oh, I hadn't considered that. Let me see if I can do something to combine them again.

File: src/app/loadout/loadout-edit/LoadoutEditBucketDropTarget.m.scss (hunk 1 above)
Comment (nev-r) [reply src/app/loadout/loadout-edit/LoadoutEditBucketDropTarget.m.scss:32]: definitely prefer it without the ghostly icons & borders, yeah

**Comment by ryan-rushton** [comment]
@nev-r I have moved over to using @bhollis's approach in #7905. It should still have the same functionality as before.

File: src/app/loadout-drawer/LoadoutDrawerContents.tsx [hunk 2]
```diff
@@ -343,8 +345,8 @@ export function fillLoadoutFromEquipped(
       item.equipped &&
       itemCanBeInLoadout(item) &&
       (category
-        ? category === 'subclass'
-          ? item.bucket.hash === BucketHashes.Subclass
+        ? category === 'General'
+          ? item.bucket.hash !== BucketHashes.Subclass && item.bucket.sort === category
```
Comment (ryan-rushton) [inline src/app/loadout-drawer/LoadoutDrawerContents.tsx:349]: I couldn't find a place where we were passing in `subclass` so I just deleted it. Now, when we are dealing with the general category we will fill everything but subclasses.

File: src/app/loadout-drawer/LoadoutDrawerContents.tsx [hunk 3]
```diff
@@ -332,6 +332,8 @@ export function fillLoadoutFromEquipped(
   itemsByBucket: { [bucketId: string]: DimItem[] },
   dimStore: DimStore,
   onUpdateLoadout: (loadout: Loadout) => void,
+  // This is a bit dangerous as it is only used from the new loadout edit drawer and
+  // has special handling that would break the old loadout drawer
```
Comment (bhollis) [inline src/app/loadout-drawer/LoadoutDrawerContents.tsx:336]: Right, that's why it's optional :-D

File: src/app/loadout-drawer/LoadoutDrawerContents.tsx (hunk 2 above)
Comment (bhollis) [reply src/app/loadout-drawer/LoadoutDrawerContents.tsx:349]: I think that's just from before I switched to `setLoadoutSubclassFromEquipped`

File: src/app/loadout/loadout-edit/LoadoutEditBucketDropTarget.m.scss (hunk 1 above)
Comment (bhollis) [reply src/app/loadout/loadout-edit/LoadoutEditBucketDropTarget.m.scss:32]:
```suggestion
  visibility: hidden;
```

Comment (bhollis) [reply src/app/loadout/loadout-edit/LoadoutEditBucketDropTarget.m.scss:32]: (you can probably remove the pointer-events too)

Comment (ryan-rushton) [reply src/app/loadout/loadout-edit/LoadoutEditBucketDropTarget.m.scss]: I will pull it down and see if I can remove it

File: src/app/loadout-drawer/LoadoutDrawerContents.tsx (hunk 3 above)
Comment (ryan-rushton) [reply src/app/loadout-drawer/LoadoutDrawerContents.tsx:336]: Yeah just figured I would call it out incase any one else that was unfamiliar came across it. I had find all the references and see where it was getting called from.

File: src/app/loadout/loadout-edit/LoadoutEditBucketDropTarget.m.scss (hunk 1 above)
Comment (ryan-rushton) [reply src/app/loadout/loadout-edit/LoadoutEditBucketDropTarget.m.scss]: Done

**Review by bhollis — APPROVED** [review]

**Pull request merged**